sql/core.go

```diff
@@ -411,6 +454,106 @@ type SystemVariable struct {
 	ValueFunction func() (interface{}, error)
 }
 
+// GetName implements SystemVariable.
+func (s *MysqlSystemVariable) GetName() string {
+	return s.Name
+}
+
+// SetName implements SystemVariable.
+func (s *MysqlSystemVariable) SetName(n string) {
+	s.Name = n
+}
+
+// GetType implements SystemVariable.
+func (s *MysqlSystemVariable) GetType() Type {
+	return s.Type
+}
+
+// SetDefault implements SystemVariable.
+func (s *MysqlSystemVariable) SetDefault(a any) {
+	s.Default = a
+}
+
+// GetDefault implements SystemVariable.
+func (s *MysqlSystemVariable) GetDefault() any {
+	return s.Default
+}
+
+// GetValue implements SystemVariable.
+func (s *MysqlSystemVariable) GetValue(a any) (any, bool) {
+	// should always be MysqlSystemVariable
+	if s.ValueFunction != nil {
+		result, err := s.ValueFunction()
+		if err != nil {
+			logrus.StandardLogger().Warnf("unable to get value for system variable %s: %s", s.Name, err.Error())
+			return nil, true
+		}
+		return result, true
+	}
+
+	// convert any set types to strings
+	if sysType, ok := s.GetType().(SetType); ok {
+		if setTypeVal, ok := a.(uint64); ok {
+			var err error
+			a, err = sysType.BitsToString(setTypeVal)
+			if err != nil {
+				return nil, false
+			}
+		}
+	}
+	return a, true
+}
+
+// HasDefaultValue implements SystemVariable.
+func (s *MysqlSystemVariable) HasDefaultValue(a any) bool {
+	return s.Default == a
+}
+
+// AssignValue implements SystemVariable.
+func (s *MysqlSystemVariable) AssignValue(val any) (SystemVarValue, error) {
+	convertedVal, _, err := s.Type.Convert(val)
+	if err != nil {
+		return SystemVarValue{}, err
+	}
+	svv := SystemVarValue{
+		Var: s,
+		Val: convertedVal,
+	}
+	if s.NotifyChanged != nil {
+		err = s.NotifyChanged(SystemVariableScope_Global, svv)
+		if err != nil {
+			return SystemVarValue{}, err
+		}
+	}
+	return svv, nil
+}
+
+// SetValue implements SystemVariable.
+func (s *MysqlSystemVariable) SetValue(val any, global bool) (SystemVarValue, error) {
+	if global && s.Scope == SystemVariableScope_Session {
+		return SystemVarValue{}, ErrSystemVariableSessionOnly.New(s.Name)
+	}
+	if !s.Dynamic || s.ValueFunction != nil {
+		return SystemVarValue{}, ErrSystemVariableReadOnly.New(s.Name)
+	}
+	return s.AssignValue(val)
+}
+
+// IsReadOnly implements SystemVariable.
+func (s *MysqlSystemVariable) IsReadOnly() bool {
+	return !s.Dynamic || s.ValueFunction != nil
+}
+
+// IsGlobalOnly implements SystemVariable.
+func (s *MysqlSystemVariable) IsGlobalOnly() bool {
+	return s.Scope == SystemVariableScope_Global
+}
+
+// GetNotifyChanged implements SystemVariable.
+func (s *MysqlSystemVariable) GetNotifyChanged() func(SystemVariableScope, SystemVarValue) error {
+	return s.NotifyChanged
+}
+
 // SystemVariableScope represents the scope of a system variable.
 type SystemVariableScope byte
```

Comment:
Perhaps make this an interface instead. That way, we can change it in Doltgres as it evolves, while not adding explicit Doltgres/Postgres aspects to GMS. That would let us remove the enum value here and only have it on the Doltgres side.